code/modules/reagents/chemistry/machinery/reagentgrinder.dm

```diff
@@ -528,3 +528,376 @@
+						O.reagents.remove_reagent("nutriment", min(O.reagents.get_reagent_amount("nutriment"), space))
+					if(O.reagents.has_reagent("plantmatter"))
+						internal.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount("plantmatter") * abs(amount) * efficiency), space))
+						O.reagents.remove_reagent("plantmatter", min(O.reagents.get_reagent_amount("plantmatter"), space))
+			else
+				O.reagents.trans_id_to(internal, r_id, min(amount, space))
+
+			if(internal.reagents.holder_full())
+				break
+
+		O.reagents.trans_to(internal, O.reagents.total_volume)
+
+		if(!O.reagents.total_volume)
+			remove_object(O)
+		if(internal.reagents.holder_full())
+			return
+
+	// Sheets and rods(!)
+	for(var/obj/item/stack/O in holdingitems)
+		var/list/special_blend = get_special_blend(O)
+		if(!length(special_blend))
+			continue // Ignore stackables that don't grind into anything
+
+		var/space = internal.reagents.maximum_volume - internal.reagents.total_volume
+		while(O.amount)
+			O.amount--
+
+			for(var/r_id in special_blend)
+				var/spaceused = min(special_blend[r_id] * efficiency, space)
+				space -= spaceused
+				internal.reagents.add_reagent(r_id, spaceused)
+
+			if(O.amount < 1) // If leftover small - destroy
+				remove_object(O)
+				break
+			if(internal.reagents.holder_full())
+				return
+
+	// Plants
+	for(var/obj/item/grown/O in holdingitems)
+		var/list/special_blend = get_special_blend(O)
+		for(var/r_id in special_blend)
+			var/space = internal.reagents.maximum_volume - internal.reagents.total_volume
+			var/amount = special_blend[r_id]
+
+			if(amount == 0)
+				if(O.reagents.has_reagent(r_id))
+					internal.reagents.add_reagent(r_id, min(O.reagents.get_reagent_amount(r_id) * efficiency, space))
+			else
+				internal.reagents.add_reagent(r_id, min(amount * efficiency, space))
+
+			if(internal.reagents.holder_full())
+				break
+
+		remove_object(O)
+		if(internal.reagents.holder_full())
+			return
+
+	// Slime Extracts
+	for(var/obj/item/slime_extract/O in holdingitems)
+		var/space = internal.reagents.maximum_volume - internal.reagents.total_volume
+
+		if(O.reagents)
+			var/amount = O.reagents.total_volume
+			O.reagents.trans_to(internal, min(amount, space))
+		if(O.Uses > 0)
+			internal.reagents.add_reagent("slimejelly", min(20 * efficiency, space))
+
+		remove_object(O)
+		if(internal.reagents.holder_full())
+			return
+
+	// Everything else - Transfers reagents from it into internal
+	for(var/obj/item/reagent_containers/O in holdingitems)
+		O.reagents.trans_to(internal, O.reagents.total_volume)
+
+		if(!O.reagents.total_volume)
+			remove_object(O)
+		if(internal.reagents.holder_full())
+			return
+
+
+//juice into the internal tank
+
+/obj/machinery/reagentgrinder/Botanitank/proc/botaniJuice()
+	power_change()
+	if(stat & (NOPOWER|BROKEN))
+		return
+	if(!internal || internal.reagents.holder_full())
+		return
+
+	playsound(src.loc, 'sound/machines/juicer.ogg', 20, 1)
```

Comment:
```suggestion
	playsound(src.loc, 'sound/machines/juicer.ogg', 20, TRUE)
```

Reply:
These are all funcitons copied from the reagent grinder section and modified a bit for the Botanitank. I think it would be better to go over the entire file and I feel like this should be a second PR. There's also an infinite beaker glitch because the regular grinder constructor just makes one each time you build it, and funnily enough you have to remove the previous beaker to deconstruct it with a screw driver and crow bar. 
Either way I think I'll fix those separately.

Reply:
Good point. I'll just go over the entire file here then. no point only half fixing it.